Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -36,22 +36,47 @@ public struct AbstractInterpreter {
     
     public mutating func reset() {
         ifStack.removeAll()
-        stack = [VariableMap<Type>()]
+        stateDiffsStack = [VariableMap<Type>()]
         propertyTypes.removeAll()
         methodSignatures.removeAll()
     }
 
-    /// Get variables with new type which have different type in old state
-    /// Used after block end when some state on stack is removed
-    private func getStateChanges(oldState: VariableMap<Type>, newState: VariableMap<Type>) -> [(Variable, Type)] {
-        var typeChanges = [(Variable, Type)]()
-        for (variable, type) in newState {
-            let oldType = oldState[variable]
-            if oldType == nil || type != oldType  {
-                typeChanges.append((variable, type))
+    /// Update current variable types and type diffs
+    /// Used after block end when some state is removed
+    private mutating func updateStateDiffs(
+        lastStateDiff: VariableMap<Type>, typeChanges: inout [(Variable, Type)], mergeTypes: Bool = true
+    ) {
+        for (variable, oldType) in lastStateDiff {
+            var newType = oldType
+
+            if mergeTypes {
+                // For now, if variable did not exist in parent state we keep its current type
+                // In future we can get rid of these types when we update types in correct time
+                // i.e. innerOutputs after updating state and outputs before updating state
+                if oldType == .nothing {
```

Comment:
As we discussed we can either follow comment and get rid of this case
or we can use analyzer which hides ceratin types when variable is out of scope